zospy/zpcore.py

```diff
@@ -300,7 +300,7 @@ def __new__(cls, *args, **kwargs):
         if len(cls._instances) >= 1:
             # As the number of applications within runtime is limited to 1 by Zemax, it is logical to also limit the
             # number of ZOS instances
-            raise ValueError(
+            raise ConnectionRefusedError(
```

Comment:
I am not sure about this one. Using a `ConnectionRefusedError` suggests that something went wrong when connecting, which simply isn't true. This method doesn't handle anything connection-related, and raising this exception here is therefore misleading. If we really want to change this, raising a `RuntimeError` seems more appropriate.

Reply:
Oke! I will change it to `RuntimeError` as I think `ValueError` was also not appropriate in hindsight.

Reply:
After discussing with @crnh, I reverted the changes of this error type as it might induce some breaking changes, thus, we will change it whenever more breaking changes will be implemented.